Cubical/Categories/Presheaf/KanExtension.agda

```diff
@@ -0,0 +1,188 @@
+
+    module _ (d : D.ob) where
+
+      Raw : Type ℓ
+      Raw = Σ[ c ∈ C.ob ] Σ[ g ∈ D.Hom[ d , F ⟅ c ⟆ ] ] G .F-ob c .fst
+
+      data _≈_ : (u v : Raw) → Type ℓ where
+        shift : {c c' : C.ob} (g : D.Hom[ d , F ⟅ c ⟆ ]) (f : C.Hom[ c , c' ]) (a : (G ⟅ c' ⟆) .fst)
+          → (c' , (g D.⋆ F ⟪ f ⟫) , a) ≈ (c , g , (G ⟪ f ⟫) a)
+
+      Quo = Raw / _≈_
+
+    pattern shift/ g f a i = eq/ _ _ (shift g f a) i
+
+    -- Action of Quo on arrows in D
+
+    mapR : {d d' : D.ob} (h : D.Hom[ d' , d ]) → Quo d → Quo d'
+    mapR h [ c , g , a ] = [ c , h D.⋆ g , a ]
+    mapR h (shift/ g f a i) =
+      hcomp
+        (λ j → λ
+          { (i = i0) → [ _ , D.⋆Assoc h g (F ⟪ f ⟫) j , a ]
+          ; (i = i1) → [ _ , h D.⋆ g , (G ⟪ f ⟫) a ]
+          })
+        (shift/ (h D.⋆ g) f a i)
+    mapR h (squash/ t u p q i j) =
+      squash/ (mapR h t) (mapR h u) (cong (mapR h) p) (cong (mapR h) q) i j
+
+    mapRId : (d : D.ob) → mapR (D.id d) ≡ (idfun _)
+    mapRId d =
+      funExt (elimProp (λ _ → squash/ _ _) (λ (c , g , a) i → [ c , D.⋆IdL g i , a ]))
+
+    mapR∘ : {d d' d'' : D.ob}
+      (h' : D.Hom[ d'' , d' ]) (h : D.Hom[ d' , d ])
+      → mapR (h' D.⋆ h) ≡ mapR h' ∘ mapR h
+    mapR∘ h' h =
+      funExt (elimProp (λ _ → squash/ _ _) (λ (c , g , a) i → [ c , D.⋆Assoc h' h g i , a ]))
+
+  LanOb : Functor (C ^op) (SET ℓ) → Functor (D ^op) (SET _)
+  LanOb G .F-ob d .fst = Quo G d
+  LanOb G .F-ob d .snd = squash/
+  LanOb G .F-hom = mapR G
+  LanOb G .F-id {d} = mapRId G d
+  LanOb G .F-seq h h' = mapR∘ G h' h
+
+  -- Action of Quo on arrows in Cᵒᵖ → Set
+
+  module _ {G G' : Functor (C ^op) (SET ℓ)} (α : NatTrans G G') where
+
+    mapL : (d : D.ob) → Quo G d → Quo G' d
+    mapL d [ c , g , a ] = [ c , g , α .N-ob c a ]
+    mapL d (shift/ g f a i) =
+      hcomp
+        (λ j → λ
+          { (i = i0) → [ _ , (g D.⋆ F ⟪ f ⟫) , α .N-ob _ a ]
+          ; (i = i1) → [ _ , g , funExt⁻ (α .N-hom f) a (~ j) ]
+          })
+        (shift/ g f (α . N-ob _ a) i)
+    mapL d (squash/ t u p q i j) =
+      squash/ (mapL d t) (mapL d u) (cong (mapL d) p) (cong (mapL d) q) i j
+
+    mapLR : {d d' : D.ob} (h : D.Hom[ d' , d ])
+      → mapL d' ∘ mapR G h ≡ mapR G' h ∘ mapL d
+    mapLR h = funExt (elimProp (λ _ → squash/ _ _) (λ _ → refl))
+
+  mapLId : (G : Functor (C ^op) (SET ℓ))
+    (d : D.ob) → mapL (idTrans G) d ≡ idfun (Quo G d)
+  mapLId G d = funExt (elimProp (λ _ → squash/ _ _) (λ _ → refl))
+
+  mapL∘ : {G G' G'' : Functor (C ^op) (SET ℓ)}
+    (β : NatTrans G' G'') (α : NatTrans G G')
+    (d : D.ob) → mapL (seqTrans α β) d ≡ mapL β d ∘ mapL α d
+  mapL∘ β α d = funExt (elimProp (λ _ → squash/ _ _) (λ _ → refl))
+
+  LanHom : {G G' : Functor (C ^op) (SET ℓ)}
+    → NatTrans G G' → NatTrans (LanOb G) (LanOb G')
+  LanHom α .N-ob = mapL α
+  LanHom α .N-hom = mapLR α
+
+  -- Definition of the left Kan extension functor
+
+  Lan : Functor (FUNCTOR (C ^op) (SET ℓ)) (FUNCTOR (D ^op) (SET (ℓ-max (ℓ-max (ℓ-max ℓC ℓC') ℓD') ℓS)))
```

Comment:
Can't you use ℓ as level here?

Reply:
Yeah, just missed that one when I introduced the abbreviation.